### PR TITLE
chore(repo): Update CODEOWNERS to require approval for versioning PRs

### PR DESCRIPTION
**Pull request by BRKalow**
## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [x] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [ ] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [ ] `@clerk/types`
- [ ] `@clerk/themes`
- [ ] `@clerk/localizations`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/fastify`
- [ ] `@clerk/chrome-extension`
- [ ] `gatsby-plugin-clerk`
- [x] `build/tooling/chore`

## Description

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.

In order to have more granular access control over who can trigger stable releases of our packages, we are adding a `CODEOWNERS` rule targeting the `CHANGELOG.md` files in each package. These files are only changed by the automated "Version Packages" PR that is generated by Changesets, so it should be a reasonable way to control access to publishing.

fixes JS-697

### CODE REVIEW

File: CODEOWNERS [hunk 1]
```diff
@@ -1,2 +1,5 @@
 * @SokratisVidros @nikosdouvlis
 packages/backend/*       @dimkl
+
+# Require specific approval for the generated "Version Packages" PR, which triggers a release
+packages/*/CHANGELOG.md @clerkinc/team-leads @BRKalow
```
Comment (BRKalow) [inline CODEOWNERS:5]:
```suggestion
packages/*/CHANGELOG.md @clerkinc/team-leads @clerkinc/team-js 
```

File: CODEOWNERS [hunk 2]
```diff
@@ -1,2 +1,5 @@
 * @SokratisVidros @nikosdouvlis
 packages/backend/*       @dimkl
+
+# Require specific approval for the generated "Version Packages" PR, which triggers a release
+packages/*/CHANGELOG.md @clerkinc/team-leads @clerkinc/team-js
```
Comment (BRKalow) [inline CODEOWNERS:5]:
```suggestion
packages/*/CHANGELOG.md @clerkinc/team-leads @clerkinc/team-js
```

**Review by nikosdouvlis — APPROVED** [review]
Great :) 
We now need to disable the JIT checks that keep failling

**Review by dimkl — APPROVED** [review]

**Comment by clerk-cookie** [comment]
This PR has been automatically locked since there has not been any recent activity after it was closed. Please open a new issue for related bugs.

**Pull request merged**